posts/inside-rust/2020-03-02-recent-future-pattern-matching-improvements.md

```diff
@@ -0,0 +1,234 @@
+---
+layout: post
+title: "Recent and future pattern matching improvements"
+author: Mazdak "Centril" Farrokhzad
+description: "Reviewing recent pattern matching improvements"
+team: the language <https://www.rust-lang.org/governance/teams/lang> and compiler <https://www.rust-lang.org/governance/teams/compiler> teams
+---
+
+[ch_6]: https://doc.rust-lang.org/book/ch06-00-enums.html
+[ch_18]: https://doc.rust-lang.org/book/ch18-00-patterns.html
+[ref_match]: https://doc.rust-lang.org/reference/expressions/match-expr.html
+[ref_pat]: https://doc.rust-lang.org/reference/patterns.html
+[ref_place]: https://doc.rust-lang.org/reference/expressions.html#place-expressions-and-value-expressions
+
+Much of writing software revolves around checking if some data has some shape ("pattern"), extracting information from it, and then reacting if there was a match. To facilitate this, many modern languages, Rust included, support what is known as "pattern matching".
+
+> If you are new to Rust or want to refresh your knowledge, you may first want to read chapters [6, Enums and Pattern Matching][ch_6] and [18, Patterns and Matching][ch_18] in the book, or read more about [`match` expressions][ref_match] and [patterns][ref_pat] in the reference.
+
+Pattern matching in Rust works by checking if a [*place*][ref_place] in memory (the "data") matches a certain *pattern*. In this post, we will look at some recent improvements to patterns soon available in stable Rust as well as some more in already available in nightly.
+
```

Comment:
```suggestion

If you are familiar with the nightly features discussed and would like to help out with the efforts to drive them to stable, jump ahead to [*How can I help?](#how-can-i-help?).

```
@Mark-Simulacrum Something like this? (Not sure the link is correct wrt. the `?`)

Reply:
Sure, that'll work.

`?` won't hurt because in the worst case it's just a `?foo` starter and it'll not bother the browser I think.